### PR TITLE
Add Documentation for lightly/models/modules

### PR DESCRIPTION
**Pull request by ayush22iitbhu**
#1687 

I have updated the docstrings for the following files:
1. lightly/models/modules/center.py
2. lightly/models/modules/heads_timm.py
3. lightly/models/modules/heads.py

### CODE REVIEW

File: lightly/models/modules/heads.py [hunk 1]
```diff
@@ -392,11 +399,10 @@ def assign_groups(self, x: Tensor) -> Tensor:
         """Assigns each representation in x to a group based on cosine similarity.
 
         Args:
-            Tensor of shape bsz x dim.
+            Tensor of shape (bsz, dim).
```
Comment (MalteEbner) [inline lightly/models/modules/heads.py:402]:
```suggestion
            x:
                Tensor of shape (bsz, dim).
```

File: lightly/models/modules/heads.py [hunk 2]
```diff
@@ -408,13 +414,19 @@ class SMoGProjectionHead(ProjectionHead):
...
 
     def __init__(
         self, input_dim: int = 2048, hidden_dim: int = 2048, output_dim: int = 128
     ):
+        """Initializes the SMoGProjectionHead with the specified dimensions.
+
+        Args:
+            input_dim: Dimensionality of the input features.
+            hidden_dim: Dimensionality of the hidden layers.
+            output_dim: Dimensionality of the output features.
```
Comment (MalteEbner) [inline lightly/models/modules/heads.py:428]:
```suggestion
            input_dim:
                 Dimensionality of the input features.
            hidden_dim:
                 Dimensionality of the hidden layers.
            output_dim:
                Dimensionality of the output features.
```

File: lightly/models/modules/heads.py [hunk 3]
```diff
@@ -435,13 +447,20 @@ class SMoGPredictionHead(ProjectionHead):
...
 
     def __init__(
         self, input_dim: int = 128, hidden_dim: int = 2048, output_dim: int = 128
     ):
+        """Initializes the SMoGPredictionHead with the specified dimensions.
+
+        Args:
+            input_dim: Dimensionality of the input features.
+            hidden_dim: Dimensionality of the hidden layers.
+            output_dim: Dimensionality of the output features.
```
Comment (MalteEbner) [inline lightly/models/modules/heads.py:461]:
```suggestion
            input_dim:
                 Dimensionality of the input features.
            hidden_dim:
                 Dimensionality of the hidden layers.
            output_dim:
                Dimensionality of the output features.
```

File: lightly/models/modules/heads.py [hunk 4]
```diff
@@ -456,13 +475,19 @@ class SimSiamPredictionHead(ProjectionHead):
...
 
     def __init__(
         self, input_dim: int = 2048, hidden_dim: int = 512, output_dim: int = 2048
     ):
+        """Initializes the SimSiamPredictionHead with the specified dimensions.
+
+        Args:
+            input_dim: Dimensionality of the input features.
+            hidden_dim: Dimensionality of the hidden layers.
+            output_dim: Dimensionality of the output features.
```
Comment (MalteEbner) [inline lightly/models/modules/heads.py:489]:
```suggestion
            input_dim:
                 Dimensionality of the input features.
            hidden_dim:
                 Dimensionality of the hidden layers.
            output_dim:
                Dimensionality of the output features.
```

File: lightly/models/modules/heads.py [hunk 5]
```diff
@@ -536,6 +563,16 @@ def __init__(
     def forward(
         self, x: Tensor, step: Optional[int] = None
     ) -> Union[Tensor, List[Tensor]]:
+        """Forward pass of the SwaVPrototypes module.
+
+        Args:
+            x: Input tensor.
+            step: Current training step.
```
Comment (MalteEbner) [inline lightly/models/modules/heads.py:570]:
```suggestion
            x:
                Input tensor.
            step:
                Current training step.
```

File: lightly/models/modules/heads.py [hunk 6]
```diff
@@ -779,6 +824,14 @@ def __init__(
         output_dim: int = 8192,
         num_layers: int = 3,
     ):
+        """Initializes the VICRegProjectionHead with the specified dimensions.
+
+        Args:
+            input_dim: Dimensionality of the input features.
+            hidden_dim: Dimensionality of the hidden layers.
+            output_dim: Dimensionality of the output features.
+            num_layers: Number of layers in the projection head.
```
Comment (MalteEbner) [inline lightly/models/modules/heads.py:833]:
```suggestion
            input_dim:
                 Dimensionality of the input features.
            hidden_dim:
                 Dimensionality of the hidden layers.
            output_dim:
                Dimensionality of the output features.
             num_layers:
                 Number of layers in the projection head.
```

File: lightly/models/modules/heads_timm.py [hunk 7]
```diff
@@ -9,6 +9,15 @@ class AIMPredictionHeadBlock(Module):
...
     - [0]: AIM, 2024, https://arxiv.org/abs/2401.08541
+
+    Args:
+        input_dim: Dimensionality of the input features.
+        output_dim: Dimensionality of the output features.
+        mlp_ratio: Ratio used to determine the hidden layer size in the MLP.
+        proj_drop: Dropout rate for the projection layer.
+        act_layer: Activation layer to use.
+        norm_layer: Normalization layer to use.
+        mlp_layer: MLP layer to use.
```
Comment (MalteEbner) [inline lightly/models/modules/heads_timm.py:20]: Update

File: lightly/models/modules/heads_timm.py [hunk 8]
```diff
@@ -41,6 +60,18 @@ class AIMPredictionHead(Module):
...
+        input_dim: Dimensionality of the input features.
+        output_dim: Dimensionality of the output features.
+        hidden_dim: Dimensionality of the hidden layer.
+        num_blocks: Number of blocks in the prediction head.
+        mlp_ratio: Ratio used to determine the hidden layer size in the MLP.
+        proj_drop: Dropout rate for the projection layer.
+        act_layer: Activation layer to use.
+        norm_layer: Normalization layer to use.
+        mlp_layer: MLP layer to use.
+        block_fn: Block function to use for the prediction head.
```
Comment (MalteEbner) [inline lightly/models/modules/heads_timm.py:74]: Update

File: lightly/models/modules/heads_timm.py [hunk 9]
```diff
@@ -79,5 +112,13 @@ def __init__(
         )
 
     def forward(self, x: Tensor) -> Tensor:
+        """Forward pass of the AIMPredictionHead.
+
+        Args:
+            x: Input tensor.
```
Comment (MalteEbner) [inline lightly/models/modules/heads_timm.py:118]: Update

**Review by MalteEbner — COMMENTED** [review]
Great work, well done!

Just one note about style: Explanations of args go in another line, see the [example file](https://github.com/lightly-ai/lightly/blob/master/lightly/models/modules/memory_bank.py).

**Comment by ayush22iitbhu** [comment]
@MalteEbner I've done the instructed changes.

**Review by MalteEbner — APPROVED** [review]
Looks perfect now, thanks a lot!

I quickly fixed the wrong indent of the docstring for the `SMoGPrototypes` and formatted the file, it is ready to merge now.

**Pull request merged**